src/util/prolog/prolog.test.ts

```diff
@@ -0,0 +1,296 @@
+				const results = (await session.executeQuery("likes(lisa, X).")).getResultArray();
+				expect(results).toStrictEqual([["bob"], ["false"]]);
+
+		});
+
+		test("Two Variables One Result", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl).", 1000);
+				const results = (await session.executeQuery("likes(lisa, X, Y).")).getResultArray();
+
+				expect(results).toStrictEqual([["bob", "karl"], ["false"]]);
+
+		});
+
+		test("Three Variables One Result", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl, friedrich).", 1000);
+				const results = (await session.executeQuery("likes(lisa, X, Y, Z).")).getResultArray();
+
+				expect(results).toStrictEqual([["bob", "karl", "friedrich"], ["false"]]);
+
+		});
+
+		test("Four Variables One Result", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl, friedrich, dieter).", 1000);
+				const results = (await session.executeQuery("likes(lisa, X, Y, Z, A).")).getResultArray();
+
+				expect(results).toStrictEqual([["bob", "karl", "friedrich", "dieter"], ["false"]]);
+
+		});
+
+		test("One Variable Two Results", async () => {
+
+				const session = new PrologSession("likes(lisa, bob). likes(lisa, dieter).", 1000);
+				const results = (await session.executeQuery("likes(lisa, X).")).getResultArray();
+				expect(results).toStrictEqual([["bob"], ["dieter"], ["false"]]);
+
+		});
+
+		test("Two Variables Two Results", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl). likes(lisa, karl, bob).", 1000);
+				const results = (await session.executeQuery("likes(lisa, X, Y).")).getResultArray();
+
+				expect(results).toStrictEqual([["bob", "karl"], ["karl", "bob"], ["false"]]);
+
+		});
+
+		test("Three Variables Two Results", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl, friedrich). likes(lisa, bob, friedrich, karl).", 1000);
+				const results = (await session.executeQuery("likes(lisa, X, Y, Z).")).getResultArray();
+
+				expect(results).toStrictEqual([["bob", "karl", "friedrich"], ["bob", "friedrich", "karl"], ["false"]]);
+
+		});
+
+		test("Four Variables Two Results", async () => {
+				const session = new PrologSession(`
+					likes(lisa, bob, karl, friedrich, dieter).
+					likes(lisa, bob, karl, dieter, friedrich).
+					`, 1000);
+				const results = (await session.executeQuery("likes(lisa, X, Y, Z, A).")).getResultArray();
+
+				expect(results).toStrictEqual(
+					[["bob", "karl", "friedrich", "dieter"], ["bob", "karl", "dieter", "friedrich"], ["false"]],
+				);
+
+		});
+
+		test("Cut Operator Maximum Function", async () => {
+
+				const session = new PrologSession("max(X,Y,Y)  :-  X  =<  Y, !. max(X, Y, X).", 1000);
+				const results = (await session.executeQuery("max(3, 7, X).")).getResultArray();
+				expect(results).toStrictEqual([["7"], ["false"]]);
+
+		});
+
+		test("Cut Operator Endless Loop", async () => {
```

Comment:
Hm, could you use sentences as a description? Usually these strings are something like `"returns an empty string if the input is invalid"` or `"returns each word only once"` or `"throws an exception if reader is not initialized"` - basically describe the expected behavior in a readable way.